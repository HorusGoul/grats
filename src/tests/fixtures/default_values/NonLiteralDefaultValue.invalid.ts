/** @gqlType */
export default class Query {
  /** @gqlField */
  hello({ greeting = String(Math.random()) }: { greeting: string }): string {
    return `${greeting} world!`;
  }
}
