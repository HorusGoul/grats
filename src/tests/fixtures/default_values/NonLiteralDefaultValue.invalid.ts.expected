-----------------
INPUT
----------------- 
/** @GQLType */
export default class Query {
  /** @GQLField */
  hello({ greeting = String(Math.random()) }: { greeting: string }): string {
    return `${greeting} world!`;
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/default_values/NonLiteralDefaultValue.invalid.ts:4:22 - error: Expected GraphQL field argument default values to be a literal.

If you think Grats should be able to infer this type, please report an issue at https://github.com/captbaritone/grats/issues.

4   hello({ greeting = String(Math.random()) }: { greeting: string }): string {
                       ~~~~~~~~~~~~~~~~~~~~~
